src/rust/src/x509/crl.rs

```diff
@@ -260,11 +260,24 @@ impl CertificateRevocationList {
 
     #[getter]
     fn extensions(&mut self, py: pyo3::Python<'_>) -> pyo3::PyResult<pyo3::PyObject> {
+        let tbs_cert_list = &self.owned.borrow_value().tbs_cert_list;
+
+        let extensions = match tbs_cert_list.extensions() {
+            Ok(exts) => exts,
+            Err(oid) => {
+                let oid_obj = oid_to_py_oid(py, &oid)?;
+                return Err(exceptions::DuplicateExtension::new_err((
+                    format!("Duplicate {} extension found", oid),
+                    oid_obj.into_py(py),
+                )));
+            }
+        };
+
```

Comment:
This shouldn't be repeated in every `extensions` method, it should be in `parse_and_cache_extensions` which can take the result of calling `extensions()` on the cert.

Reply:
Making sure I understand: in that case, `parse_and_cache_extensions` should take a `Result<Option<Extensions>>`? Because that'll be the return type from `extensions()`.

(Or are you thinking it should take the `RawExtensions`, and then convert it to an `Extensions` internally? That would avoid passing a potential `Err` variant through as a parameter.)

Reply:
I've done this with `RawExtensions`; the error handling is now pushed fully into `parse_and_cache_extensions`.